backend/api/Services/AutoScheduleService.cs

```diff
@@ -226,6 +226,10 @@ public async Task AutoScheduleMissionRun(string missionDefinitionId, TimeOnly ti
                 return;
             }
 
+            var pendingMissionRuns = await missionRunService.ReadMissionRunQueue(robot.Id);
+            if (pendingMissionRuns.Any((m) => m.MissionId == missionDefinition.Id))
+                return; // No need to schedule already scheduled mission
```

Comment:
Add logging before return so its easier to understand behavior

Reply:
Done 👍